src/Sylius/Behat/Context/Setup/CustomerContext.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class CustomerContext implements Context
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $customerFactory;
+
+    /**
+     * @param RepositoryInterface $customerRepository
+     * @param SharedStorageInterface $sharedStorage
+     * @param FactoryInterface $customerFactory
+     */
+    public function __construct(
+        RepositoryInterface $customerRepository,
+        SharedStorageInterface $sharedStorage,
+        FactoryInterface $customerFactory
+    ) {
+        $this->customerRepository = $customerRepository;
+        $this->sharedStorage = $sharedStorage;
+        $this->customerFactory = $customerFactory;
+    }
+
+    /**
+     * @Then there should be customer with email :email
```

Comment:
Only `Given` steps should not operate on UI and should be in the `Setup` namespace. The rest of them should be done through UI.